javascript/javascript2/week1/spirit_animal/script.js

```diff
@@ -0,0 +1,73 @@
+//Spirit animal name generator
+document.querySelectorAll('.radio').forEach(radio => {
+   radio.addEventListener('click', function(event){
+      const usernameInput = document.getElementById('username');
+      const buttonSpiritAnimal = document.getElementById('btnSpiritAnimal');
+
+      buttonSpiritAnimal.removeEventListener('click', handleButtonClick);
+      usernameInput.removeEventListener('mouseover', handleMouseOver);
+      usernameInput.removeEventListener('input', handleTextInput);
+      
+      if (event.target.value === 'btnClick') {
+         buttonSpiritAnimal.addEventListener('click', handleButtonClick);
+      } else if (event.target.value === 'hoverInput') {
+         usernameInput.addEventListener('mouseover', handleMouseOver);
+      } else if(event.target.value === 'textInput') {
+         usernameInput.addEventListener('input', handleTextInput);
+      }
+   })
+})
+
+document.getElementById('btnNewSpiritAnimal').addEventListener('click', handleNewSpiritAnimalClick);
+
+function handleButtonClick(event) {
+   handleEvent(event, 'result');
+}
+
+function handleNewSpiritAnimalClick(event) {
+   handleEvent(event, 'result2');
+}
+
+function handleMouseOver(event) {
+   handleEvent(event, 'result');
+}
+
+function handleTextInput(event) {
+   handleEvent(event, 'result');
+}
```

Comment:
Great approach of extracting the common logic into a shared function. 

Right now you have 3 functions all doing the exact same thing. you could consider to define a single function and reuse that all 3 places. What would a fitting name for such a function be?